crates/net/network/src/transactions/fetcher.rs

```diff
@@ -1072,13 +1081,45 @@ impl Default for TransactionFetcher {
     }
 }
 
+#[derive(Debug, Clone)]
+pub(super) struct TxSizeMetadata {
+    pub peer_id: PeerId,
+    pub tx_encoded_len: usize,
+}
+
+impl PartialEq for TxSizeMetadata {
+    fn eq(&self, other: &Self) -> bool {
+        self.peer_id == other.peer_id
+    }
+}
+
+impl Eq for TxSizeMetadata {}
+
+impl Hash for TxSizeMetadata {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.peer_id.hash(state);
```

Comment:
i think this is unexpected behavior, so not sure i am a fan. it should at least be documented